examples/run_perf_test_reads_2.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+"""
+Simple script to get performance profile of read throughput.
+
+This script accepts a single argument `-e=SCALE` as a power of 10.
+
+-e=2 is equivalent to 500 records.
+-e=3 is equivalent to 5_000 records.
+-e=4 is equivalent to 50_000 records.
+-e=5 is equivalent to 500_000 records.
+-e=6 is equivalent to 5_000_000 records.
+
+Use smaller values of `e` (2-3) to understand read and overhead costs.
+Use larger values of `e` (4-5) to understand write throughput at scale.
+
+For performance profiling, use `viztracer` to generate a flamegraph:
+```
+poetry run viztracer --open -- ./examples/run_perf_test_reads.py -e=3
+poetry run viztracer --open -- ./examples/run_perf_test_reads.py -e=5
+```
+
+To run without profiling, prefix script name with `poetry run python`:
+```
+# Run with 5_000 records
+poetry run python ./examples/run_perf_test_reads.py -e=3
+# Run with 500_000 records
+poetry run python ./examples/run_perf_test_reads.py -e=5
+
+# Load 5_000 records to Snowflake
+poetry run python ./examples/run_perf_test_reads.py -e=3 --cache=snowflake
+
+# Load 5_000 records to BigQuery
+poetry run python ./examples/run_perf_test_reads.py -e=3 --cache=bigquery
+```
+
+Note:
+- For purposes of estimating MB/s throughput, each record of the 'purchases' stream is
+  assumed to be 220 bytes, meaning 4_500 records is approximately 1 MB.
+- With these estimates, 20K records/second is approximately 4.5 MB/s,
+  and 25K records/second is approximately 5.5 MB/s.
```

Comment:
Stale, from copy-paste. Needs to be updated with the sizes of the e2e sources' stream.